griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+import logging
+from schema import Schema, Literal, Optional
+from attr import define, field
+from griptape.artifacts import ErrorArtifact, InfoArtifact, ListArtifact, BlobArtifact
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient
+from googleapiclient.errors import HttpError
+from google.auth.exceptions import MalformedError
+from googleapiclient.discovery import Resource
+
+@define
+class GoogleDriveClient(BaseGoogleClient):
+    owner_email: str = field(kw_only=True)
+    LIST_FILES_SCOPES = ['https://www.googleapis.com/auth/drive.readonly']
+    UPLOAD_FILE_SCOPES = ['https://www.googleapis.com/auth/drive.file']
+    max_files: int = field(default=10, kw_only=True)
+
+    def _build_client(self, scopes: list[str]) -> Resource:
+        from google.oauth2 import service_account
+        from googleapiclient.discovery import build
+        credentials = service_account.Credentials.from_service_account_info(
+            self.service_account_credentials, scopes=scopes
+        )
+        delegated_credentials = credentials.with_subject(self.owner_email)
+        service = build('drive', 'v3', credentials=delegated_credentials)
+        return service
+
+    def list_files(self, params: dict) -> ListArtifact | ErrorArtifact:
```

Comment:
What happened to `@activities` for all of the methods?

Reply:
I just removed them as of now, wanted to check with you on how to write them correctly.